ring-core/src/ring/middleware/params.clj

```diff
@@ -8,25 +8,32 @@
   (let [params (codec/form-decode params encoding)]
     (if (map? params) params {})))
 
+(defn- assoc-param-map [req k v]
+  (some-> req (assoc k (if-let [v' (req k)]
+                         (reduce-kv assoc v' v)
+                         v))))
+
 (defn assoc-query-params
   "Parse and assoc parameters from the query string with the request."
   {:added "1.3"}
   [request encoding]
-  (merge-with merge request
-    (if-let [query-string (:query-string request)]
-      (let [params (parse-params query-string encoding)]
-        {:query-params params, :params params})
-      {:query-params {}, :params {}})))
+  (let [params (if-let [query-string (:query-string request)]
+                 (parse-params query-string encoding)
+                 {})]
+    (-> request
+        (assoc-param-map :query-params params)
+        (assoc-param-map :params params))))
 
 (defn assoc-form-params
   "Parse and assoc parameters from the request body with the request."
   {:added "1.2"}
   [request encoding]
-  (merge-with merge request
-    (if-let [body (and (req/urlencoded-form? request) (:body request))]
-      (let [params (parse-params (slurp body :encoding encoding) encoding)]
-        {:form-params params, :params params})
-      {:form-params {}, :params {}})))
+  (let [params (if-let [body (and (req/urlencoded-form? request) (:body request))]
```

Comment:
Can you ensure this line doesn't exceed 80 characters?

Reply:
Done